R/youth_allowance.R

```diff
@@ -0,0 +1,198 @@
+            if_else(has_children,
+                    if_else(is_single,
+                            1440.50,
+                            1280.34),
+                    if_else(living_at_home,
+                            if_else(age<18,
+                                    864.84,
+                                    948.50),
+                            1206.17)),
+            #jobseeker
+            if_else(isjspceoalfofcoahodeoc,
+                    1435.17,      
+                    if_else(has_children,
+                            if_else(is_single,
+                                    1149.84,
+                                    989.67),
+                            if_else(living_at_home,
+                                    if_else(age<18,
+                                            574.18,
+                                            657.84),
+                                    989.67))))
+            
+
+
+  # Benefit test
+  # http://guides.dss.gov.au/guide-social-security-law/4/2/2
+  # https://www.humanservices.gov.au/individuals/enablers/personal-income-test-austudy-and-youth-allowance/30411
+  
+  
+  #ordinary_income_free_area <- 
+  #  cpi_inflator(100, from_fy = "2014-15", to_fy = fy.year)
+  
+  #ordinary_income_excess <- pmaxC(ordinary_income - ordinary_income_free_area)
+  
+  ###TESTS needed: https://www.humanservices.gov.au/individuals/services/centrelink/youth-allowance-students-and-australian-apprentices/how-much-you-can-get/income-and-assets-test#personalassets
+    #personal income test: function ya_income_reduction()
+      #https://www.humanservices.gov.au/individuals/enablers/personal-income-test-austudy-and-youth-allowance/30411
+  
+    #personal assets test: INCOMPLETE
+      #http://guides.dss.gov.au/guide-social-security-law/4/2/8/30 The personal assets test for independent YA recipients is the same as the benefits assets tes
+      #http://guides.dss.gov.au/guide-social-security-law/4/2/3
+      #https://www.legislation.gov.au/Details/C2018C00167
+  
+    #partner income test: function ya_partner_reduction() INCOMPLETE
+      #http://guides.dss.gov.au/guide-social-security-law/4/2/2
+      #example: http://guides.dss.gov.au/guide-social-security-law/5/5/3/30
+  
+    #parental income test: function ya_parental_income_reduction() INCOMPLETE
+      #https://www.humanservices.gov.au/individuals/services/centrelink/youth-allowance-students-and-australian-apprentices/how-much-you-can-get/income-and-assets-test#parentalincome
+      #http://guides.dss.gov.au/guide-social-security-law/4/2/8/05                          
+      #family tax benefits: http://guides.dss.gov.au/family-assistance-guide/3/1/1/20   
+      #https://www.humanservices.gov.au/individuals/enablers/how-much-ftb-part-you-can-get/30611
+  
+                                       
+  payment <- if_else(ordinary_income > max_income,
+                     0,
+                     max_rate_March_2018 - ya_income_reduction(ordinary_income, 
```

Comment:
I think this construction is fine, but the variable name needs to be (eventually) changed so that it can be applied to other dates.